test/native-promises.tap.js

```diff
@@ -0,0 +1,619 @@
+if (!global.Promise) return;
```

Comment:
Are there any versions of node/io that might be broken in their promise support because of half baked implementations? Such as 0.11 with `--harmony`, if so, do we care?